mycompany-seeq-link-connector-myconnector/src/main/java/com/mycompany/seeq/link/connector/MyConnection.java

```diff
@@ -174,29 +186,49 @@ public void index(SyncMode syncMode) {
             // If you need the signals to be written to Seeq Server before any other work continues, you can
             // call flushSignals() on the connection service.
             this.connectionService.putSignal(signal);
+
+            ConditionUpdateInputV1 condition = new ConditionUpdateInputV1();
+
+            // The Data ID is a string that is unique within the data source, and is used by Seeq when referring
+            // to condition data. Data ID is a string and does not need to be numeric, even though we are just
+            // using a number in this example.
+            condition.setDataId(String.format("%d", tag.getId()));
+
+            // The Name is a string that is displayed in the UI. It can change (typically as a result of a
+            // rename operation happening in the source system), but the unique Data ID preserves appropriate
+            // linkages.
+            condition.setName(tag.getName());
+
+            // PutCondition() queues items up for performance reasons and writes them in batch to the server.
+            //
+            // If you need the conditions to be written to Seeq Server before any other work continues, you can
+            // call FlushConditions() on the connection service.
+            this.connectionService.putCondition(condition);
+
+            this.createChildAsset(rootAssetId, signal.getDataId(), signal.getName());
```

Comment:
I think usually we create the asset tree first then signals and conditions (top-down)